Marlin/src/HAL/STM32/HAL.h

```diff
@@ -123,7 +123,7 @@
 // Types
 // ------------------------
 
-typedef int16_t pin_t;
+typedef uint32_t pin_t;
```

Comment:
Is it really a 32-bit integer now? That seems excessive, and might make the code a bit larger and slower than it needs to be.

Reply:
It's not *explicitly* defined, but I was getting overflow conversion errors from PinName to pin_t

https://github.com/stm32duino/Arduino_Core_STM32/blob/master/cores/arduino/stm32/PinNames.h

I think it has to do with the splitting on digital and analog pins. They went with `uint32_t` for analog for some unknown reason...

https://github.com/stm32duino/Arduino_Core_STM32/blob/47097af5039c90ff634cd7beb040b5b1f7074c14/variants/STM32F4xx/F446R(C-E)T/variant_generic.cpp#L71